File: digital-cricket.cpp
#include<bits/stdc++.h>
using namespace std;
void swap(int*a, int*b)
{
	int temp = *a;
	*a = *b;
	*b = temp;
}
int main()
{
	int N; 
	cin>>N;
	int balls[N];
	for(int i = 0; i<N; i++)
	{
		cin>>balls[i];
	}
	int striker = 1, non_striker  = 2,max_score = INT_MIN;
	unordered_map<int,int> map;
	for(int i = 0; i<N; i++)
	{
         if((i+1)%6 == 0)
         {
         	swap(&striker,&non_striker);
         }
         if(balls[i] == 0 || balls[i] == -1)
         {
         	map[striker]+=balls[i];
         	max_score = max(max_score, map[striker]);
         	striker = max(striker, non_striker)+1;
         }
         else if(balls[i] == 4 || balls[i] == 6 || balls[i] == 2)
         {
         	map[striker]+=balls[i];
         }
         else if(balls[i] == 1 || balls[i] == 3)
         {
         	map[striker]+=1;
         	swap(&striker, &non_striker);
         }
	}
	max_score = max(max_score,max(map[striker], map[non_striker]));
	int player_pos = INT_MAX, top_scorer;
	for(auto it : map)
	{
		if(it.second == max_score)
		{
              if(it.first < player_pos)
              {
              	top_scorer = it.first;
              	player_pos = it.first;
              }
		}
	}
	cout<<top_scorer<<" "<<striker;
}

File: League-1_contest-3/digital-cricket.py
n = int(input())
lis = list(map(int,input().split()))
max_score = 0;
index = -1;
(striker, non_striker,dict) = (1,2,{})
for i in range(1,12):
    dict[i] = 0;
for i in lis:
    index+=1;
    if(i == 0 or i == -1):
            max_score = max(max_score,dict[striker]);
            striker = max(striker,non_striker)+1;
    elif(i == 1 or i == 3):
         dict[striker]+=i;
         (striker,non_striker) = (non_striker,striker)
    else:
         dict[striker]+=i;
    if((index+1)%6 == 0):
         (striker,non_striker) = (non_striker,striker)
max_score = max(max_score, max(dict[striker], dict[non_striker]));
print(max_score)
min_index = 1000000
top_scorer = 0;
for key, value in dict.items():
    if(value == max_score):
        if(key < min_index):
            top_scorer = key;
            min_index = key;
print(top_scorer,striker)

       
    
    
        
